files/default/bin/deploy.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+PUBLIC=0
+
+run_help() {
+  echo
+  echo " NAME:"
+  echo "    $0 - deploy target image to OpenStack for further testing."
+  echo
+  echo " USAGE:"
+  echo "    $0 [-p | -f [FILE]] -n [IMG_NAME] -r [PR_NUM]"
+  echo
+  echo " OPTIONS:"
+  echo "    -p          - Publish (publicly) the target image as '\$IMG_NAME', renaming the old"
+  echo "                  '\$IMG_NAME' to '\$IMG_NAME - \$PR_NUM' and making that image private"
+  echo
+  echo " ARGUMENTS:"
+  echo "    FILE        - Fully built image to be uploaded to OpenStack if [-p] is not"
+  echo "                  specified. The image name will be: '\$IMG_NAME - \$PR_NUM'."
+  echo
+  echo "    IMG_NAME    - Prefix of the new image or name of the image to replace if"
+  echo "                  [-p] is specified."
+  echo
+  echo "    PR_NUM      - GitHub Pull Request number to suffix private images with."
+  echo
+  exit 0
+}
+
+while getopts "hpf:n:r:" opt ; do
+  case ${opt} in
+    h)
+      run_help
+      ;;
+    p)
+      PUBLIC=1
```

Comment:
This variable is confusing as you describe it as "publish" in the help section and yet here you use "public". My vote is to keep it publish throughout so it's less confusing.